Switch to libuv 1.4.2 on new libuv home

muojp opened the pull request:
Made this PR regarding the discussion: https://github.com/aspnet/aspnet-docker/issues/21

I also performed a quick test against latest libuv 1.1.0 and it worked fine. Maybe we shall jump to 1.1.0 instead of 1.0.2. Regarding [1.1.0 changelog](https://github.com/libuv/libuv/releases/tag/v1.1.0), there are no significant changes (especially for unix env.), so in my opinion skipping 1.0.2 and move on to 1.1.0 is relatively safe.

ahmetalpbalkan commented:
@davidfowl please merge. :shipit:

Eilon commented:
We need to update Kestrel before accepting this I think. Kestrel is ultimately the only component that uses libuv.

akoeplinger commented:
@Eilon it shouldn't matter as 1.x is compatible (SemVer ftw :smile:)

Eilon commented:
Understood; I'm just saying we need to update Kestrel as well so that we can be consistent in which version of libuv we're using.

akoeplinger commented:
@Eilon yeah, that makes sense. Opened https://github.com/aspnet/KestrelHttpServer/issues/50 to track it.

Eilon commented:
Thanks. Waiting for @lodejard to return from vacation to look at these.

muojp commented:
Update: modified this PR to cover 1.0.0-beta2

ahmetalpbalkan commented:
@lodejard pinging again to see if we can upgrade image to libuv 1.1.0.

ahmetalpbalkan commented:
@lodejard pinging again to see if we can upgrade image to libuv 1.1.0 in the docker image

akoeplinger commented:
The latest stable is now 1.4.2 already :smile:

ahmetalpbalkan commented:
I believe we can rely on manual testing + CI to update this. @muojp frel free to uodate PR with latest libuv release. Sorry for being slow on this.

muojp commented:
@ahmetalpbalkan sure. I'll update the PR in several hours.

muojp commented:
@ahmetalpbalkan @lodejard
Hi, I've just finished testing 1.0.0-beta2+libuv-1.4.2 manually on my box and updated the PR.

Hope https://github.com/aspnet/aspnet-docker/pull/38 will be also updated once this PR is merged.

Also, maybe it's about time for us to prepare some policies for stopping maintenance of older betas for focusing on keeping up with latest ones instead. Making middle-wares used on older betas partially up to date sometimes cause painful troubles on users (tho I don't see there are any atm). In my opinion, we could limit performing maintenance against (N-1)-th & N-th beta (N=current) which is suitable for most users. Of course, this could be argued in another issue.

ahmetalpbalkan commented:
@muojp can you update beta3 as well?

muojp commented:
@ahmetalpbalkan Updated to cover beta 3. :cat:

ahmetalpbalkan commented:
@muojp You rock. :metal:
